### PR TITLE
Restructure configuration pages

### PR DESCRIPTION
**Pull request by mgnslndh**
@pascalberger:

> I would suggest to have docs/configuration/ giving a brief introduction to configuration for Cake, plus the list of sub-pages (like e.g. done at https://cakebuild.net/docs/extending/modules/). This would be in the index.cshtml file. I think the most part of docs/fundamentals/configuration.md can be moved to something like docs/configuration/set-configuration-values.md.

Part of #907

### CODE REVIEW

File: input/docs/running-builds/configuration/index.cshtml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Order: 40
+RedirectFrom: docs/running-builds/configuration
```
Comment (pascalberger) [inline input/docs/running-builds/configuration/index.cshtml:2]: This redirects to itself and leads to an infinite redirect

**Review by pascalberger — CHANGES_REQUESTED** [review]
There something wrong with one of the redirects. Otherwise LGTM

**Comment by gep13** [comment]
Shouldn't the existing redirects remain in place? So that both old locations for the pages continue to get to the right place?

**Comment by pascalberger** [comment]
> Shouldn't the existing redirects remain in place? So that both old locations for the pages continue to get to the right place?

Yes missed that one. We need to keep existing redirects and add additional redirects for the move in this PR

File: input/docs/running-builds/configuration/index.cshtml (hunk 1 above)
Comment (mgnslndh) [reply input/docs/running-builds/configuration/index.cshtml:2]: Yes, that does not seem right. I'll fix it.

**Review by pascalberger — APPROVED** [review]
LGTM

**Comment by pascalberger** [comment]
@mgnslndh your changes have been merged, thanks for your contribution 👍

**Pull request merged**